router/router.go

```diff
@@ -141,8 +146,7 @@ func (r *routeSet) ProcessRoute(uri *url.URL) (newurl string, requirePermission
 }
 
 func (r *routeSet) GetProxy(req *ms_http.Request) (*url.URL, error) {
-	route := r.match(req.URL)
-	if route == nil || !route.target.useProxy {
+	if !r.hostHash["http://"+req.URL.Host] {
```

Comment:
This won't work for https-based targets.
You must use the base url here. You may also need to parse the URL, and then use protocol + hostname (eg "http://www.example.com". My recommendation is actually to base it on the hostname proper, but if you do that, then you need to properly parse the URLs wherever you're using this stuff - you can't just mix and mix full URLs with hostnames, etc. Changing this matching system to use proper hostnames invalidates my above comment about changing the variable away from using the term "host" in it.